drgn_tools/rds.py

```diff
@@ -1020,6 +1117,421 @@ def rds_sock_info(
+        lport_list = fields_to_list(lport)
+    if rport:
+        rport_list = fields_to_list(rport)
+
+    rds_sock_list_p = prog["rds_sock_list"].address_of_()
+
+    for rs in list_for_each_entry(
+        "struct rds_sock", rds_sock_list_p, "rs_item"
+    ):
+        for inc in list_for_each_entry(
+            "struct rds_incoming", rs.rs_recv_queue.address_of_(), "i_item"
+        ):
+            lport_rm = be16_to_host(prog, inc.i_hdr.h_dport)
+            rport_rm = be16_to_host(prog, inc.i_hdr.h_sport)
+            seq = be64_to_host(prog, inc.i_hdr.h_sequence)
+            num_bytes = be32_to_host(prog, inc.i_hdr.h_len)
+            local_addr = rds_inet_ntoa(rs.rs_bound_sin6.sin6_addr)
+            remote_addr = rds_inet_ntoa(inc.i_saddr)
+            tos_rm = int(inc.i_conn.c_tos)
+
+            if laddr and local_addr not in laddr_list:
+                continue
+            if raddr and remote_addr not in raddr_list:
+                continue
+            if tos and str(tos_rm) not in tos_list:
+                continue
+            if lport and str(lport_rm) not in lport_list:
+                continue
+            if rport and str(rport_rm) not in rport_list:
+                continue
+
+            index = index + 1
+            msg_queue.append(
+                [
+                    index,
+                    local_addr,
+                    lport_rm,
+                    remote_addr,
+                    rport_rm,
+                    tos_rm,
+                    seq,
+                    num_bytes,
+                ]
+            )
+
+    print("Receive Message Queue:")
+    print_table(msg_queue, outfile, report)
+    print("\n")
+    return None
+
+
+def rds_print_send_retrans_msg_queue(
+    prog: drgn.Program,
+    queue: str,
+    laddr: Optional[str] = None,
+    raddr: Optional[str] = None,
+    tos: Optional[str] = None,
+    lport: Optional[str] = None,
+    rport: Optional[str] = None,
+    ret: Optional[bool] = False,
+    outfile: Optional[str] = None,
+    report: bool = False,
+) -> None:
+    """
+    Print the rds send or retransmit msg queue similar rds-info -st
+
+    :param prog: drgn program
+    :prarm queue: The msg queue to be displaed. Ex: 'send', 'recv'
+    :param laddr: comma seperated string list of LOCAL-IP.  Ex: '192.168.X.X, 10.211.X.X, ...'
+    :param raddr: comma seperated string list of REMOTE-IP.  Ex: '192.168.X.X, 10.211.X.X, ...'
+    :param tos: comma seperated string list of TOS.  Ex: '0, 3, ...'
+    :param lport: comma seperated string list of lport.  Ex: 2259, 36554, ...'
+    :param rport: comma seperated string list of rport.  Ex: 2259, 36554, ...'
+    :param outfile: A file to write the output to.
+    :param report: Open the file in append mode. Used to generate a report of all the functions in the rds module.
+    :returns: None
+
+    """
+    msg = ensure_debuginfo(prog, ["rds"])
+    if msg:
+        print(msg)
+
+    index = -1
+    msg_queue: List[List[Any]] = [
```

Comment:
Would be nice to use `Table()` here as well, following the same general approac above.